sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGeneratorTest.java

```diff
@@ -48,6 +49,54 @@ public void binaryOperatorPriorityPreserved() {
             "(", parameterNames.get(1), CriteriaType.OR.getSqlKeyword(), parameterNames.get(2), ")"));
     }
 
+    @Test
+    public void generateBinaryQueryWithStartsWithDoesNotUseUpper() {
+        Criteria nameStartsWith = Criteria.getInstance(CriteriaType.STARTS_WITH, "firstName",
+            Collections.singletonList("TREVOR"),
+            Part.IgnoreCaseType.ALWAYS);
+        CosmosQuery query = new CosmosQuery(nameStartsWith);
+
+        SqlQuerySpec result = queryGenerator.generateCosmos(query);
+
+        Assert.assertEquals(result.getQueryText(), " WHERE STARTSWITH(r.firstName, @firstName0, true) ");
+    }
+
+    @Test
+    public void generateBinaryQueryWithArrayContainsUsesUpper() {
+        Criteria hasLastName = Criteria.getInstance(CriteriaType.ARRAY_CONTAINS, "lastName",
+            Collections.singletonList("ANDERSON"),
+            Part.IgnoreCaseType.ALWAYS);
+        CosmosQuery query = new CosmosQuery(hasLastName);
+
+        SqlQuerySpec result = queryGenerator.generateCosmos(query);
+
+        Assert.assertEquals(result.getQueryText(), " WHERE ARRAY_CONTAINS(UPPER(r.lastName), UPPER(@lastName0)) ");
+    }
+
+    @Test
+    public void generateBinaryQueryWithIsEqualIntUsesUpper() {
+        Criteria isEqualInt = Criteria.getInstance(CriteriaType.IS_EQUAL, "zipcode",
+            Collections.singletonList(20180),
+            Part.IgnoreCaseType.ALWAYS);
+        CosmosQuery query = new CosmosQuery(isEqualInt);
+
+        SqlQuerySpec result = queryGenerator.generateCosmos(query);
+
+        Assert.assertEquals(result.getQueryText(), " WHERE UPPER(r.zipcode) = UPPER(@zipcode0) ");
```

Comment:
I am confused.
If I am having a document like following:
```
{
    "id": "id",
    "mypk": "mypk1",
    "someNumber": 1123
}
```
select * from r where UPPER(r. someNumber) = UPPER(1123) will return nothing?

Reply:
This test case is for
```
{
    "id": "id",
    "mypk": "mypk1",
    "someNumber": "1123"
}
```

Reply:
"real" numbers wouldn't hit the getFunctionCondition code path but getCondition - and never be converted into UPPER

Reply:
ah okie, makes a lot sense now, thanks :)